### PR TITLE
set http_proxy/https_proxy for systemd environment

### PR DESCRIPTION
**Pull request by menghan**

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -255,7 +255,7 @@
     regexp: "^export HTTP_PROXY="
     line: "export HTTP_PROXY=\"{{docker_http_proxy}}\""
     state: present
-  when: docker_http_proxy is defined and (docker_http_proxy != None)
+  when: docker_http_proxy is defined and (docker_http_proxy != None) and not is_systemd
```
Comment (lhoss) [inline tasks/main.yml]: I propose to replace this check (and others similarly) by:
```
  when:
  - (docker_http_proxy|default('')) != ''
  - not is_systemd
```
which is a cleaner check for `docker_http_proxy` (that I originally contributed), as I just came across a case where the check didn't work (as expected, on v2.3.1), when defining the variable like this:
```
docker_http_proxy: "{{ proxy_env.http_proxy | default(None) }}"
```

File: tasks/main.yml [hunk 2]
```diff
@@ -126,7 +126,7 @@
     owner: root
     group: root
     mode: 0755
-  when: docker_opts != "" and is_systemd
+  when: (docker_opts != "" or (docker_http_proxy is defined and (docker_http_proxy != None)) or (docker_https_proxy is defined and (docker_https_proxy != None))) and is_systemd
```
Comment (lhoss) [inline tasks/main.yml]: see my other comment , this will also make the check shorter

**Comment by lhoss** [comment]
+1 to get proxy with systemd support merged
todo resolve conflict

**Comment by lhoss** [comment]
> @menghan if you fix the comments from @lhoss and resolve the conflicts, I'll get this merged
>  👍 1

yeah that would be perfect (else I can take over)
also I can spare myself  the mini fix PR (fixing the `(docker_http_proxy|default('')) != ''`  check on the master branch)

**Comment by angstwad** [comment]
@lhoss I've added you as a collaborator.  Go ahead and take care of this if you want.

**Comment by lhoss** [comment]
> @lhoss I've added you as a collaborator. Go ahead and take care of this if you want.

FIXED the conflict as proposed 
and tested that it works for the existing (non systemd) case, both with a proxy set, and without

**Pull request merged**